src/hardware/mixer.cpp

```diff
@@ -1324,13 +1325,48 @@ std::unique_ptr<Program> MIXER_ProgramCreate() {
 	return ProgramCreate<MIXER>();
 }
 
-void MIXER_Init(Section* sec) {
+bool MIXER_IsManuallyMuted()
+{
+	return mixer.state == MixerState::Mute;
+}
+
+void MIXER_SetState(const MixerState requested)
+{
+	assert(requested != MixerState::Uninitialized);
+
+	// Already in the requested state or the NoSound state?
+	if (mixer.state == requested || mixer.state == MixerState::NoSound ||
+	    !mixer.sample_rate.load())
```

Comment:
What's `!mixer.sample_rate.load()`?

Reply:
Because `MIXER_SetState()` is now tied to SDL eventing - we can get calls into it even before the Mixer's come to life.  So this was an initial check for that - but it's wasn't direct.

I now directly check the `sounddevice`  ID value - because the SetState() needs to use it in the call to either pause or unpause the audio device.  I'm a lot happier w/ it now, and it's robust against all the startup orders that we can throw at it.